Hire-Wire/hire-wire
# Apply fixes to UserController response, fixes to LLMController to correctly return a response

**Update UserController and Fixes for LLM-Related Services**

**What does this PR do?**

This PR updates the structure of the response object in `UserController` to include user ID information after login or registration. Additionally, it includes necessary fixes for `LLMController`, `LLMGenerationService`, and `jobApplicationCreationService` to ensure the LLM successfully generates and returns a response.

**List of tasks completed**

-   [x] Updated `UserController` response structure to include `userId` in front-end response.
-   [x] Adjusted unit tests for `UserController` to reflect updated response format.
-   [x] Resolved issues in `LLMController` to ensure proper response handling.
-   [x] Applied fixes to `LLMGenerationService` and `jobApplicationCreationService` to support successful LLM responses.

**How can this be tested?**

1. Run unit tests for `UserController` to verify response structure.
2. Test LLM functionality by triggering content generation and checking for successful responses.

**Additional info**

N/A
